src/main/scala/Repository.scala

```diff
@@ -1,44 +1,49 @@
 import spray.json._
 import DefaultJsonProtocol._
+import sangria.execution.deferred.{DeferredResolver, Fetcher, Relation, RelationIds}
 
-object Repository {
-  println("Data loaded:")
+import scala.concurrent.Future.successful
 
-  val categorySource = io.Source.fromFile("src/main/resources/categories.json")("UTF-8").mkString.parseJson
-  implicit val categoryFormat = jsonFormat2(Category.apply)
-  val categories = categorySource.convertTo[List[Category]]
+class Repository extends Fetchers {
+  val categories = loadFile[Category]("categories.json")
+  val styles = loadFile[Style]("styles.json")
+  val breweries = loadFile[Brewery]("breweries.json")
+  val beers = loadFile[Beer]("beers.json")
 
-  println(s"  - ${categories.length} categories")
-
-  val styleSource = io.Source.fromFile("src/main/resources/styles.json")("UTF-8").mkString.parseJson
-  implicit val styleFormat = jsonFormat3(Style.apply)
-  val styles = styleSource.convertTo[List[Style]]
+  private def loadFile[T : JsonFormat](fileName: String): List[T] =
+    io.Source.fromResource(fileName)("UTF-8").mkString.parseJson.convertTo[List[T]]
+}
```

Comment:
Ah! I did try exactly that but I got confused with the required `implicit` definition of `jsonFormatN`, brilliant!